include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -139,10 +139,29 @@ MultigridPreconditionerBase<dim, Number>::initialize_levels(unsigned int const d
 
   std::vector<unsigned int> h_levels;
 
+
   // setup h-levels
-  unsigned int const n_h_levels = get_number_of_h_levels();
-  for(unsigned int h = 0; h < n_h_levels; h++)
-    h_levels.push_back(h);
+  if(data.involves_h_transfer())
```

Comment:
Note that the old code version from PR #451 did not use the member function `involves_h_transfer()`, and that the present PR changes this member function!

@peterrum could you therefore please check that the logic is now as intended?